modules/Media.js

```diff
@@ -24,31 +24,51 @@ class Media extends React.Component {
 
   queries = [];
 
-  state = {
-    matches:
-      this.props.defaultMatches ||
-      Object.keys(this.props.queries).reduce(
-        (acc, key) => ({ ...acc, [key]: true }),
-        {}
-      )
-  };
+  constructor(props) {
+    super(props);
+
+    if (typeof window !== "object") {
+      // In case we're rendering on the server
+      this.state = {
+        matches:
+          this.props.defaultMatches ||
+          Object.keys(this.props.queries).reduce(
+            (acc, key) => ({ ...acc, [key]: true }),
+            {}
+          )
+      };
+      return;
+    }
 
-  updateMatches = () => {
-    const newMatches = this.queries.reduce(
+    this.initialize();
+
+    // Instead of calling this.updateMatches, we manually set the state to prevent
+    // calling setState, which could trigger an unnecessary second render
+    this.state = {
+      matches:
+        this.props.defaultMatches !== undefined
+          ? this.props.defaultMatches
+          : this.getMatches()
+    };
+    this.onChange();
```

Comment:
I'm unsure whether we should be calling `onChange` in the constructor. I added it now because this is in line with the current behavior.